pkgs/tools/networking/network-manager/default.nix

```diff
@@ -16,44 +18,34 @@ in stdenv.mkDerivation rec {
     sha256 = "0b2x9hrg41cd17psqi0vacwj733v99hxczn53gdfs0yanqrji5lf";
   };
 
-  outputs = [ "out" "dev" ];
-
-  postPatch = ''
-    patchShebangs ./tools
-  '';
-
-  preConfigure = ''
-    substituteInPlace configure --replace /usr/bin/uname ${coreutils}/bin/uname
-    substituteInPlace configure --replace /usr/bin/file ${file}/bin/file
-
-    # Fixes: error: po/Makefile.in.in was not created by intltoolize.
-    intltoolize --automake --copy --force
-  '';
+  outputs = [ "out" "dev" "devdoc" "man" "doc" ];
 
   # Right now we hardcode quite a few paths at build time. Probably we should
   # patch networkmanager to allow passing these path in config file. This will
   # remove unneeded build-time dependencies.
-  configureFlags = [
-    "--with-dhclient=${dhcp}/bin/dhclient"
-    "--with-dnsmasq=${dnsmasq}/bin/dnsmasq"
+  mesonFlags = [
+    "-Ddhclient=${dhcp}/bin/dhclient"
+    "-Ddnsmasq=${dnsmasq}/bin/dnsmasq"
     # Upstream prefers dhclient, so don't add dhcpcd to the closure
-    "--with-dhcpcd=no"
-    "--with-pppd=${ppp}/bin/pppd"
-    "--with-iptables=${iptables}/bin/iptables"
+    "-Ddhcpcd=no"
+    "-Dpppd=${ppp}/bin/pppd"
+    "-Diptables=${iptables}/bin/iptables"
     # to enable link-local connections
-    "--with-udev-dir=${placeholder "out"}/lib/udev"
-    "--with-resolvconf=${openresolv}/sbin/resolvconf"
-    "--sysconfdir=/etc" "--localstatedir=/var"
-    "--with-dbus-sys-dir=${placeholder "out"}/etc/dbus-1/system.d"
-    "--with-crypto=gnutls" "--disable-more-warnings"
-    "--with-systemdsystemunitdir=$(out)/etc/systemd/system"
-    "--with-kernel-firmware-dir=/run/current-system/firmware"
-    "--with-session-tracking=systemd"
-    "--with-modem-manager-1"
-    "--with-nmtui"
-    "--disable-gtk-doc"
-    "--with-libnm-glib" # legacy library, TODO: remove
-    "--disable-tests"
+    "-Dudev_dir=${placeholder "out"}/lib/udev"
+    "-Dresolvconf=${openresolv}/sbin/resolvconf"
+    "-Ddbus_conf_dir=${placeholder "out"}/etc/dbus-1/system.d"
+    "-Dsystemdsystemunitdir=${placeholder "out"}/etc/systemd/system"
+    "-Dkernel_firmware_dir=/run/current-system/firmware"
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
+    "-Dcrypto=gnutls"
+    "-Dsession_tracking=systemd"
+    "-Dmodem_manager=true"
+    "-Dnmtui=true"
+    "-Ddocs=true"
+    "-Dlibnm_glib=true" # legacy library, TODO: remove
```

Comment:
It is hard to determine since programs may use the typelib at runtime, but since GNOME depended on it until 3.26 (IIRC), I would think all the GNOME forks like MATE will still require it.